No Network Crashes

Stack traces:
```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:422)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$201(ScheduledThreadPoolExecutor.java:152)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:265)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1112)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:587)
	at java.lang.Thread.run(Thread.java:818)
Caused by: rx.exceptions.OnErrorNotImplementedException: Unable to resolve host "www.thebluealliance.com": No address associated with hostname
	at rx.Observable$27.onError(Observable.java:7535)
	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:154)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:111)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:197)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:170)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 7 more
Caused by: java.net.UnknownHostException: Unable to resolve host "www.thebluealliance.com": No address associated with hostname
	at java.net.InetAddress.lookupHostByName(InetAddress.java:457)
	at java.net.InetAddress.getAllByNameImpl(InetAddress.java:252)
	at java.net.InetAddress.getAllByName(InetAddress.java:215)
	at com.squareup.okhttp.internal.Network$1.resolveInetAddresses(Network.java:29)
	at com.squareup.okhttp.internal.http.RouteSelector.resetNextInetSocketAddress(RouteSelector.java:187)
	at com.squareup.okhttp.internal.http.RouteSelector.next(RouteSelector.java:2156)
	at com.squareup.okhttp.internal.http.HttpEngine.createNextConnection(HttpEngine.java:344)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:2327)
	at com.squareup.okhttp.Call.getResponse(Call.java:267)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:224)
	at com.thebluealliance.androidclient.datafeed.APIv2RequestInterceptor.intercept(APIv2RequestInterceptor.java:45)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:221)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:195)
	at com.squareup.okhttp.Call.access$100(Call.java:34)
	at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:162)
	at com.squareup.okhttp.internal.NamedRunnable.run(NamedRunnable.java:33)
	... 3 more
Caused by: android.system.GaiException: android_getaddrinfo failed: EAI_NODATA (No address associated with hostname)
	at libcore.io.Posix.android_getaddrinfo(Native Method)
	at libcore.io.ForwardingOs.android_getaddrinfo(ForwardingOs.java:55)
	at java.net.InetAddress.lookupHostByName(InetAddress.java:438)
	... 18 more
```

```
java.lang.IllegalStateException: Exception thrown on Scheduler.Worker thread. Add `onError` handling.
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:60)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:423)
	at java.util.concurrent.FutureTask.run(FutureTask.java:237)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:269)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1113)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:588)
	at java.lang.Thread.run(Thread.java:818)
Caused by: rx.exceptions.OnErrorNotImplementedException: SSL handshake timed out
	at rx.Observable$27.onError(Observable.java:7535)
	at rx.observers.SafeSubscriber._onError(SafeSubscriber.java:154)
	at rx.observers.SafeSubscriber.onError(SafeSubscriber.java:111)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber.pollQueue(OperatorObserveOn.java:197)
	at rx.internal.operators.OperatorObserveOn$ObserveOnSubscriber$2.call(OperatorObserveOn.java:170)
	at rx.internal.schedulers.ScheduledAction.run(ScheduledAction.java:55)
	... 6 more
Caused by: java.net.SocketTimeoutException: SSL handshake timed out
	at com.android.org.conscrypt.NativeCrypto.SSL_do_handshake(Native Method)
	at com.android.org.conscrypt.OpenSSLSocketImpl.startHandshake(OpenSSLSocketImpl.java:324)
	at com.squareup.okhttp.Connection.connectTls(Connection.java:235)
	at com.squareup.okhttp.Connection.connectAndSetOwner(Connection.java:14199)
	at com.squareup.okhttp.OkHttpClient$1.connectAndSetOwner(OkHttpClient.java:128)
	at com.squareup.okhttp.internal.http.HttpEngine.sendRequest(HttpEngine.java:2328)
	at com.squareup.okhttp.Call.getResponse(Call.java:267)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:224)
	at com.thebluealliance.androidclient.datafeed.APIv2RequestInterceptor.intercept(APIv2RequestInterceptor.java:45)
	at com.squareup.okhttp.Call$ApplicationInterceptorChain.proceed(Call.java:221)
	at com.squareup.okhttp.Call.getResponseWithInterceptorChain(Call.java:195)
	at com.squareup.okhttp.Call.access$100(Call.java:34)
	at com.squareup.okhttp.Call$AsyncCall.execute(Call.java:162)
	at com.squareup.okhttp.internal.NamedRunnable.run(NamedRunnable.java:33)
	... 3 more
```

--- a/android/src/main/java/com/thebluealliance/androidclient/binders/EventTabBinder.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/binders/EventTabBinder.java
@@ -1,9 +1,11 @@
 package com.thebluealliance.androidclient.binders;
 
+import com.thebluealliance.androidclient.Constants;
 import com.thebluealliance.androidclient.fragments.EventsByWeekFragment;
 import com.thebluealliance.androidclient.models.EventWeekTab;
 
 import android.support.annotation.Nullable;
+import android.util.Log;
 
 import java.util.List;
 
@@ -46,6 +48,7 @@ public class EventTabBinder extends AbstractDataBinder<List<EventWeekTab>> {
 
     @Override
     public void onError(Throwable throwable) {
-
+        Log.e(Constants.LOG_TAG, "Error fetching event years");
+        throwable.printStackTrace();
     }
 }

--- a/android/src/main/java/com/thebluealliance/androidclient/subscribers/YearsParticipatedDropdownSubscriber.java
+++ b/android/src/main/java/com/thebluealliance/androidclient/subscribers/YearsParticipatedDropdownSubscriber.java
@@ -1,16 +1,20 @@
 package com.thebluealliance.androidclient.subscribers;
 
 import com.google.gson.JsonArray;
+
+import com.thebluealliance.androidclient.Constants;
 import com.thebluealliance.androidclient.interfaces.YearsParticipatedUpdate;
+
+import android.util.Log;
 
 import java.util.Arrays;
 
 import javax.inject.Inject;
 
+import rx.Subscriber;
 import rx.android.schedulers.AndroidSchedulers;
-import rx.functions.Action1;
 
-public class YearsParticipatedDropdownSubscriber implements Action1<JsonArray> {
+public class YearsParticipatedDropdownSubscriber extends Subscriber<JsonArray> {
 
     private final YearsParticipatedUpdate mCallback;
 
@@ -20,7 +24,18 @@ public class YearsParticipatedDropdownSubscriber implements Action1<JsonArray> {
     }
 
     @Override
-    public void call(JsonArray apiYears) {
+    public void onCompleted() {
+
+    }
+
+    @Override
+    public void onError(Throwable e) {
+        Log.e(Constants.LOG_TAG, "Error fetching team years");
+        e.printStackTrace();
+    }
+
+    @Override
+    public void onNext(JsonArray apiYears) {
         int[] years = new int[apiYears.size()];
         for (int i = apiYears.size() - 1; i >= 0; i--) {
             years[i] = apiYears.get(i).getAsInt();

--- a/android/src/test/java/com/thebluealliance/androidclient/subscribers/YearsParticipatedDropdownSubscriberTest.java
+++ b/android/src/test/java/com/thebluealliance/androidclient/subscribers/YearsParticipatedDropdownSubscriberTest.java
@@ -31,13 +31,13 @@ public class YearsParticipatedDropdownSubscriberTest {
 
     @Test(expected = NullPointerException.class)
     public void testParseNullData() {
-        mSubscriber.call(null);
+        mSubscriber.onNext(null);
     }
 
     @Test
     public void testParsedData() {
         int[] expected = {2015, 2014, 2013, 2012};
-        mSubscriber.call(mYearsParticipated);
+        mSubscriber.onNext(mYearsParticipated);
         verify(mCallback).updateYearsParticipated(expected);
     }
 }

